x/ccv/provider/keeper/hooks.go

```diff
@@ -60,7 +60,15 @@ func (h Hooks) AfterUnbondingInitiated(ctx sdk.Context, ID uint64) {
 // Define unimplemented methods to satisfy the StakingHooks contract
 func (h Hooks) AfterValidatorCreated(ctx sdk.Context, valAddr sdk.ValAddress) {
 }
-func (h Hooks) AfterValidatorRemoved(ctx sdk.Context, _ sdk.ConsAddress, valAddr sdk.ValAddress) {
+func (h Hooks) AfterValidatorRemoved(ctx sdk.Context, ca sdk.ConsAddress, valAddr sdk.ValAddress) {
+	// Delete any key assignments that are associated with this
+	// validator across all consumer chains.
+	h.k.IterateConsumerChains(ctx, func(ctx sdk.Context, chainID, clientID string) (stop bool) {
```

Comment:
True, I have left that out of the MVP because the proposal timing stuff in general is still using the temporary solution of allowing assignments at any time.